### PR TITLE
Support pid, port, reference, and identifier types

### PR DESCRIPTION
**Pull request by yuezato**
Question: which one of the following is the correct definition?
1. `sup(TyPid, TyPort) = TyPid | TyPort`
2. `sup(TyPid, TyPort) = TyIdentifier`

### CODE REVIEW

**Comment by yoshihiro503** [comment]
It's OK: 
`sup(TyPid, TyPort) = TyPid | TyPort`

**Comment by yuezato** [comment]
Finally, we decide to let Fialyzer not have the type Identifier internally.

This decision stands on http://erlang.org/doc/reference_manual/typespec.html, which says `identifier()` is the alias of `pid() | port() | reference()`.

**Comment by yoshihiro503** [comment]
Add some testing cases.

**Comment by yuezato** [comment]
Add the following BIFs
- [ ] `self() -> pid()` http://erlang.org/doc/man/erlang.html#self-0
- [ ] `list_to_port(String) -> port()` http://erlang.org/doc/man/erlang.html#list_to_port-1
- [ ] `make_ref() -> reference()` http://erlang.org/doc/man/erlang.html#make_ref-0

Note: `list_to_port` may throw the exception `badarg`.

File: lib/type.ml [hunk 1]
```diff
@@ -264,9 +281,12 @@ let rec of_absform = function
...
      of_elem (TyList (of_elem TyNumber))
-  | F.TyPredef {name="pid"; args=[]; _}
-  | F.TyPredef {name="port"; args=[]; _}
-  | F.TyPredef {name="reference"; args=[]; _}
+  | F.TyPredef {name="pid"; args=[]; _} ->
+     of_elem TyPid
+  | F.TyPredef {name="port"; args=[]; _} ->
+     of_elem TyPort
+  | F.TyPredef {name="reference"; args=[]; _} ->
+     of_elem TyReference
```
Comment (yoshihiro503) [inline lib/type.ml]: Add the case for `identifier()` type in specs.
```suggestion
     of_elem TyReference
   | F.TyPredef {name="identifier"; args=[]; _} ->
     TyUnion [TyPid; TyPort; TyReference]
```

**Pull request merged**